green/test/test_integration.py

```diff
@@ -1,29 +1,30 @@
 import copy
 import multiprocessing
 import os
-from pathlib import PurePath
+import pathlib
+import shutil
 import subprocess
 import sys
 import tempfile
 from textwrap import dedent
 import unittest
-from unittest.mock import MagicMock
-
-from green import cmdline
 
 
 class TestFinalizer(unittest.TestCase):
-    def setUp(self):
-        self.tmpdir = tempfile.mkdtemp()
+    def setUp(self) -> None:
+        self.tmpdir = pathlib.Path(tempfile.mkdtemp())
+
+    def tearDown(self) -> None:
+        shutil.rmtree(self.tmpdir, ignore_errors=True)
 
-    def test_finalizer(self):
+    def test_finalizer(self) -> None:
         """
-        Test that the finalizer works on Python 3.8+
+        Test that the finalizer works on all supported versions of Python.
```

Comment:
We only raised the minimum supported version to 3.6.

```suggestion
        Test that the finalizer works on Python 3.8+.
```